File: app/src/main/java/ai/elimu/kukariri/BaseApplication.java
package ai.elimu.kukariri;

import android.app.Application;
import android.content.Intent;
import android.util.Log;

import ai.elimu.kukariri.service.ScreenOnService;
import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        Log.i(getClass().getName(), "onCreate");
        super.onCreate();

        // Start service for registering ScreenOnReceiver
        Intent screenOnServiceIntent = new Intent(getApplicationContext(), ScreenOnService.class);
        startService(screenOnServiceIntent);
    }
}

File: app/src/main/java/ai/elimu/kukariri/BaseApplication.kt
package ai.elimu.kukariri

import ai.elimu.kukariri.service.ScreenOnService
import android.app.Application
import android.content.Intent
import android.util.Log
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class BaseApplication : Application() {
    override fun onCreate() {
        Log.i(javaClass.name, "onCreate")
        super.onCreate()

        // Start service for registering ScreenOnReceiver
        val screenOnServiceIntent = Intent(applicationContext, ScreenOnService::class.java)
        startService(screenOnServiceIntent)
    }
}
